internal/graph/lookup.go

```diff
@@ -94,7 +85,7 @@ func (cl *ConcurrentLookup) LookupViaReachability(ctx context.Context, req Valid
 	cancelCtx, checkCancel := context.WithCancel(ctx)
 	defer checkCancel()
 
-	checker := NewParallelChecker(cancelCtx, cl.c, req.Subject, cl.concurrencyLimit)
+	checker := newParallelChecker(cancelCtx, cl.c, req, cl.concurrencyLimit)
 	stream := &collectingStream{checker, req, cancelCtx, 0, 0, 0, sync.Mutex{}}
 
 	// Start the checker.
```

Comment:
To add in the comment below: clarify that `DispatchReachableResources` is blocking until the all elements are streamed (and thus there is no race with `Wait` closing the channel)

Reply:
Did so and have the parallel checker cancel the context if it hits the requested number of results

Reply:
Did so and have the parallel checker cancel the context if it hits the requested number of results